applications/console_text_messenger/src/main.rs

```diff
@@ -214,8 +217,29 @@ pub fn main() {
     let secret_key = CommsSecretKey::from_hex(settings.secret_key.unwrap().as_str()).unwrap();
     let public_key = CommsPublicKey::from_secret_key(&secret_key);
 
-    // TODO Use a less hacky crate to determine the local machines public IP address. This only works on Unix systems!
-    let ip = local_ip::get().expect("Could not determine local machines public IP address");
+    let mut address_count = 0;
+    let mut ip = String::new();
+    // loop through network interfaces
+    let interfaces = datalink::interfaces();
+    for interface in interfaces.into_iter() {
+        // check for address that is not loopback
+        if interface.is_loopback() == false {
```

Comment:
```suggestion
        if !interface.is_loopback() {
```
is the preferred syntax

Reply:
Thanks.